src/components/RichTextEditor/state/toolbar/machine.ts

```diff
@@ -1,38 +1,52 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { EditorState } from "draft-js";
-import { createMachine, DoneInvokeEvent, sendParent } from "xstate";
+import { Editor } from "slate";
+import { createMachine, DoneInvokeEvent } from "xstate";
+import { BlockElement, FormattedText } from "../../RichTextEditor";
+import { updateBlockType, updateInlineStyle } from "./actions";
 
 export type ToolbarContext = {
-    editorState: EditorState;
+    selectedText: string;
 };
 
-export type ToolbarStateData = {
-    editorState: EditorState;
+export type BlockTypeData = {
+    active: boolean;
+    type: BlockElement["type"];
+    editor: Editor;
 };
 
+export type InlineStyleData = {
+    style: Omit<keyof FormattedText, "text">;
+    value: boolean;
+    editor: Editor;
+};
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type ToolbarData = InlineStyleData | BlockTypeData;
+
 export enum States {
     Initial = "initial",
     LinkChooser = "link_chooser",
```

Comment:
Maybe add a Todo comment to update the state machine as soon as the link chooser is implemented.
Right now the LinkChooser State isn't used anywhere.

Reply:
0c3f6e4